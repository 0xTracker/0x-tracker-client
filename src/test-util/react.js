import { createMemoryHistory } from 'history';
import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import React from 'react';

function renderWithRouter(
  element,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {},
) {
  return {
    ...render(<Router history={history}>{element}</Router>),
    history,
  };
}

export { renderWithRouter };
