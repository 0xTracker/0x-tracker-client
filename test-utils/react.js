import { createMemoryHistory } from 'history';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { render } from 'react-testing-library';
import { Router } from 'react-router-dom';
import React from 'react';

import reducer from '../src/redux/reducer';

function renderWithRouter(
  element,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {},
) {
  return {
    ...render(<Router history={history}>{element}</Router>),
    history,
  };
}

const renderWithAppContext = (
  component,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
    initialState,
    store = createStore(reducer, initialState),
  } = {},
) => ({
  ...render(
    <Provider store={store}>
      <Router history={history}>{component}</Router>
    </Provider>,
  ),
  history,
});

export { renderWithAppContext, renderWithRouter };
