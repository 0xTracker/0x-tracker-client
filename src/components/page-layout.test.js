import { render } from '@testing-library/react';
import React from 'react';

import PageLayout from './page-layout';

describe('page layout component', () => {
  it('should render with children', () => {
    const { container } = render(<PageLayout>Hello World</PageLayout>);

    expect(container.firstChild).toMatchSnapshot();
  });
});
