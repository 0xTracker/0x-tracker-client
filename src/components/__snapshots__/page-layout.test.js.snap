// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page layout component should render with children 1`] = `
<div
  class="page-layout__StyledPageLayout-cls0ob-1 erAgYD"
>
  <div
    class="container__StyledContainer-cj53s7-0 fLJuGb page-layout__PageBody-cls0ob-0 kCvweN container-fluid"
  >
    Hello World
  </div>
</div>
`;
